frontend/src/pages/modelServing/screens/metrics/MetricsChart.tsx

```diff
@@ -59,38 +63,52 @@ const MetricsChart: React.FC<MetricsChartProps> = ({
   domain = defaultDomainCalculator,
   toolbar,
   type = MetricsChartTypes.AREA,
+  isStack = false,
 }) => {
   const bodyRef = React.useRef<HTMLDivElement>(null);
   const [chartWidth, setChartWidth] = React.useState(0);
   const { currentTimeframe, lastUpdateTime } = React.useContext(ModelServingMetricsContext);
   const metrics = useStableMetrics(unstableMetrics, title);
+  const CursorVoronoiContainer = React.useMemo(() => createContainer('voronoi', 'cursor'), []);
+  const [tooltipDisabled, setTooltipDisabled] = React.useState(false);
+  const [tooltipTitle, setTooltipTitle] = React.useState(
+    convertTimestamp(Date.now(), formatToShow(currentTimeframe)),
+  );
 
   const {
     data: graphLines,
     maxYValue,
     minYValue,
+    maxXValue,
+    minXValue,
   } = React.useMemo(
     () =>
       metrics.reduce<ProcessedMetrics>(
         (acc, metric) => {
           const lineValues = createGraphMetricLine(metric);
-          const newMaxValue = Math.max(...lineValues.map((v) => v.y));
-          const newMinValue = Math.min(...lineValues.map((v) => v.y));
+          const newMaxYValue = Math.max(...lineValues.map((v) => v.y));
+          const newMinYValue = Math.min(...lineValues.map((v) => v.y));
+          const newMaxXValue = Math.max(...lineValues.map((v) => v.x));
+          const newMinXValue = Math.min(...lineValues.map((v) => v.x));
 
           return {
-            data: [...acc.data, lineValues],
-            maxYValue: Math.max(acc.maxYValue, newMaxValue),
-            minYValue: Math.min(acc.minYValue, newMinValue),
+            data: [...acc.data, { points: lineValues, name: metric.name }],
+            maxYValue: Math.max(acc.maxYValue, newMaxYValue),
+            minYValue: Math.min(acc.minYValue, newMinYValue),
+            maxXValue: Math.max(acc.maxXValue, newMaxXValue),
+            minXValue: Math.min(acc.minXValue, newMinXValue),
           };
         },
-        { data: [], maxYValue: 0, minYValue: 0 },
+        { data: [], maxYValue: 0, minYValue: 0, maxXValue: 0, minXValue: Date.now() },
```

Comment:
because it's the minimum value and we will update it when doing the reduce function. If we set it to 0, it will be never updated because all the x value will be greater than 0